File: cpp/alerts/tracker_warning.cpp
#include "./tracker_warning.h"

namespace ltrs {
    rust::Str tracker_warning_alert_get_warning_message(lt::tracker_warning_alert *a) {
        return a->warning_message();
    }

    uint8_t tracker_warning_alert_get_version(lt::tracker_warning_alert *a) {
        return static_cast<uint8_t>(a->version);
    }
}

File: src/ffi/alerts/tracker_warning.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type tracker_warning_alert = crate::ffi::ffi::tracker_warning_alert;
    }
    unsafe extern "C++" {
        include!("cpp/alerts/tracker_warning.h");

        unsafe fn tracker_warning_alert_get_warning_message<'a>(
            alert: *mut tracker_warning_alert,
        ) -> &'a str;

        unsafe fn tracker_warning_alert_get_version(alert: *mut tracker_warning_alert) -> u8;
    }
}
